### PR TITLE
Add trace log level

### PR DESCRIPTION
**Pull request by rodyt**
This PR accomplishes the following:

- Adds a "trace" log level for miscellaneous logs, as requested in #1547 
- Adds some documentation for the Ryujinx.Common.Logging module
- Refactors some logging code, most notably, removing the `PrintStub` method 

I didn't document everything in the module either because the field/class/method was self-explanatory or I didn't understand what it did well enough to feel confident in documenting it. Any help is appreciated.

### CODE REVIEW

File: Ryujinx.Common/Logging/LogEventArgs.cs [hunk 1]
```diff
@@ -6,24 +6,19 @@ public class LogEventArgs : EventArgs
...
-            Message    = message;
-        }
-
-        public LogEventArgs(LogLevel level, TimeSpan time, string threadName, string message, object data)
+        public LogEventArgs(LogLevel level, TimeSpan time, string threadName, LogClass logClass, string caller, string message, object data)
         {
             Level      = level;
             Time       = time;
             ThreadName = threadName;
+            Class = logClass;
```
Comment (riperiperi) [inline Ryujinx.Common/Logging/LogEventArgs.cs:20]: Alignment.

File: Ryujinx.Memory/Tracking/MemoryTracking.cs [hunk 2]
```diff
@@ -75,7 +75,6 @@ public void Map(ulong va, ulong pa, ulong size)
                 {
                     VirtualRegion region = results[i];
                     region.RecalculatePhysicalChildren();
-                    region.UpdateProtection();
```
Comment (riperiperi) [inline Ryujinx.Memory/Tracking/MemoryTracking.cs:78]: All changes in https://github.com/Ryujinx/Ryujinx/pull/1672 seems to be reverted (or just not rebased, IDK)

**Review by riperiperi — CHANGES_REQUESTED** [review]
https://github.com/Ryujinx/Ryujinx/pull/1672 seems to have been reverted somehow, and there is an alignment issue with 2 statements.

Worth noting that there are a ton of PRs in the horizon project that this will conflict with, but that'll be the case at all times so it's unavoidable.

Not sure if you want to make any other changes with regards to the discussions 2 days ago. I appreciate the addition of XML comments. 😌

**Comment by gdkchan** [comment]
Closing because the author doesn't intend to work on it anymore.

**Pull request closed without merging**